packages/story-editor/src/components/canvas/frameElement.js

```diff
@@ -79,42 +79,31 @@ const EmptyFrame = styled.div`
 const NOOP = () => {};
 
 function FrameElement({ element }) {
-  const { setEditingElement } = useCanvas((state) => ({
-    setEditingElement: state.actions.setEditingElement,
-  }));
+  const setEditingElement = useCanvas((v) => v.actions.setEditingElement);
   const { id, type, flip } = element;
   const { Frame, isMaskable, Controls } = getDefinitionForType(type);
   const elementRef = useRef();
   const [hovering, setHovering] = useState(false);
-  const {
-    state: { isAnythingTransforming },
-  } = useTransform();
+  const isAnythingTransforming = useTransform(
+    (v) => v.state.isAnythingTransforming
+  );
 
   const onPointerEnter = () => setHovering(true);
   const onPointerLeave = () => setHovering(false);
 
-  const { setNodeForElement, handleSelectElement, isEditing } = useCanvas(
-    (state) => ({
-      setNodeForElement: state.actions.setNodeForElement,
-      handleSelectElement: state.actions.handleSelectElement,
-      isEditing: state.state.isEditing,
-    })
-  );
-  const { selectedElementIds, currentPage, isAnimating } = useStory(
-    (state) => ({
-      selectedElementIds: state.state.selectedElementIds,
-      currentPage: state.state.currentPage,
-    })
+  const setNodeForElement = useCanvas((v) => v.actions.setNodeForElement);
+  const handleSelectElement = useCanvas((v) => v.actions.handleSelectElement);
+  const isEditing = useCanvas((v) => v.state.isEditing);
```

Comment:
Here, too, perhaps leave a comment to explain why these are using `useCanvas` separately here to ensure someone in the future won't just move these back together.

Reply:
I'll post in the slack channel going over this. I don't think we need specific comments everywhere we're doing this, but maybe a group sync and like a little performance README. What are your thoughts?